src/modules/blog/blog.service.ts

```diff
@@ -0,0 +1,116 @@
+import { NotFoundException } from './../../shared/exceptions/http.exception';
+import { Repository } from 'typeorm';
+import {
+  BadRequestException,
+  InternalServerErrorException,
+} from '../../shared/exceptions/http.exception';
+import { Blog } from './blog.entity';
+import { CreateBlogDto } from './dto/create-blog.dto';
+import { UpdateBlogDto } from './dto/update-blog.dto';
+import { v4 as uuidv4 } from 'uuid';
+
+
+export class BlogService {
+  constructor(private blogRepository: Repository<Blog>) {}
+
+  async createBlog(createBlogDto: CreateBlogDto) {
+    try {
+      const { authorId, title, content, tags } = createBlogDto;
+
+      const existingBlog = await this.blogRepository.findOne({ where: { title } });
```

Comment:
I don't think it is not required to add check for titles. You can find different articles with the same titles.